api.md

```diff
@@ -0,0 +1,86 @@
+# Proposed webrtc webextension api
+
+The initial version of the webrtc webextension api is a limited
+set of events, just enough to implement the extension from
+[this mozhacks article](https://hacks.mozilla.org/2015/09/controlling-webrtc-peerconnections-with-an-extension/).
+
+To use any of these events, an extension must have the `"webrtc"`
+permission (in addition to the `"experiments.webrtc"` permission unless/until
+this api lands in central).  In addition, an extension must have the
+`"webrtc.blocking"` permission to add a blocking onPeerConnection listener.
+
+The events are:
+
+## onPeerConnection
+
+This event is dispatched whenever an RTCPeerConnection is established.
+Usage is similar to other webextension events, though note that optional
+extra argument to `addListener()`:
+
+```js
+function listener(origin, id) { ... }
+
+browser.webrtc.onPeerConnection.addListener(listener, blocking);
+browser.webrtc.onPeerConnection.removeListener(listener);
+browser.webrtc.onPeerConnection.hasListener(listener);
+```
+
+The provided callback function (`listener` above) is passed the origin
+of the page that is opening a peer connection as a string, and an
+opaque identifier that can be used to identify the specific connection
+request being canceled when handling the 
+[`onPeerConnectionCancel`](#onPeerConnectionCancel) event.
+
+If a truthy value is passed for the `blocking` argument,
+the value returned from the listener function controls whether
+connection setup is allowed to proceed or not.  If the listener returns
+a string with the value `"allow"` or `"deny"`, the connection is allowed
+or denied, respectively.  If any other string value or other primitive
+data type is returned, the system default is applied (which typically
+allows the connection).  If the listener returns a Promise, processing
+of the connection is delayed until the Promise is resolved, at which point
+the processing described above is applied to the Promise resolution value.
+
+## onPeerConnectionCancel
```

Comment:
Right, so these are named with permission prompts in mind, so they're really `onPeerConnectionCancelPermissionRequest` if you will (not saying you should call it that, but maybe), and probably only fire if block was used, to give you a chance to remove a permission prompt before the page is unloaded.